requirements.txt

```diff
@@ -1,4 +1,3 @@
 numpy
 scipy
-matplotlib
```

Comment:
as with Jupyter, it would make sense to define sub-packages/requirements. otherwise we will loose track what the full requirements are, e.g. to also have the widged GUI and the plotting via MPL.

You can for example add [`extras`](http://peak.telecommunity.com/DevCenter/setuptools#declaring-extras-optional-features-with-their-own-dependencies) such as `GUI` and `plot` to `setup.py`:
```python
setup(
    # [...]
    extras_require = {
        'GUI':  ["ipywidgets>=A.B"],
        'plot': ["matplotlib>=X.Y"],
    }
)
```
([Newer doc](http://setuptools.readthedocs.io/en/latest/setuptools.html#declaring-extras))

We can also define an (pip install) `openPMD-viewer[all]` install target that will call to install the GUI and the plot (MPL) deps as well for proper packaging. See [the logic in IPython's setup](https://github.com/ipython/ipython/blob/dae349b90e0fe3ad37c08792f33d4b2d35dabc7c/setup.py#L172-L232) for example.

Or in a more manual case just:
```python
setup(
    # [...]
    extras_require = {
        'GUI':  ["ipywidgets>=A.B"],
        'plot': ["matplotlib>=X.Y"],
        'all': ["ipywidgets>=A.B", "matplotlib>=X.Y"],
    }
)
```